functions.py

```diff
@@ -1,36 +1,35 @@
 import time
+
+def ReadLines(nameoffile):
+    with open(nameoffile,'r') as f:
+        arrayline = f.readlines()

```

Comment:
we are relying on the fact that state is sorted by time. Let's add check that this holds.